FinniversKit/Sources/Components/BlockUserView/BlockUserView.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  BlockUserView.swift
+//  FinniversKit
+//
+//  Copyright © 2021 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public protocol BlockUserViewDelegate: AnyObject {
+    func blockUserViewDidTapCancel()
+    func blockUserViewDidTapBlock(reason: Int)
+    func blockUserViewDidTapLink()
+}
+
+public class BlockUserView: UIView {
+
+    // MARK: - Private properties
+
+    private lazy var title: Label = {
+        let label = Label(style: .title1)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private lazy var subtitle: Label = {
+        let label = Label(style: .body)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        return label
+    }()
+
+    lazy var radio: RadioButton = {
+        let radio = RadioButton()
+        radio.translatesAutoresizingMaskIntoConstraints = false
+        radio.delegate = self
+        return radio
+    }()
+
+    private lazy var info: Label = {
+        let label = Label(style: .caption)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var link: Button = {
+        let button = Button(style: .link)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setContentCompressionResistancePriority(.required, for: .horizontal)
+        button.addTarget(self, action: #selector(linkAction), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var cancel: Button = {
+        let button = Button(style: .default)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setContentCompressionResistancePriority(.required, for: .horizontal)
+        button.addTarget(self, action: #selector(cancelAction), for: .touchUpInside)
+        return button
+    }()
+
+    private lazy var block: Button = {
+        let button = Button(style: .callToAction)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(blockAction), for: .touchUpInside)
+        button.isEnabled = false
+        return button
+    }()
+
+    public weak var delegate: BlockUserViewDelegate?
+
+    private var viewModel: BlockUserViewModel? {
+        didSet {
+            title.text = viewModel?.title
+            subtitle.text = viewModel?.subtitle
+            radio.fields = viewModel?.reasons ?? []
```

Comment:
It shouldn't happen I guess, but maybe we should enable the button if there are no reasons? Like if the backend is down users will still be able to block other users even if we will not get to know why

Reply:
should not happen. but its good to go